### DISCUSSION TITLE
"Error: No wallet supplied from which to retrieve identity label".

### DISCUSSION THREAD

**Discussion by is0356xi** [hyperledger/fabric#2455] [General]
I want to copy the wallet credentials and create another wallet.
However, I am getting the error "Error: No wallet supplied from which to retrieve identity label".
What should I do?

The following source code can be executed normally.

```:
const wallet = await buildWallet(Wallets, walletPath);
await enrollAdmin(this.caClient, wallet, mspOrg1);

const gateway = new Gateway();
await gateway.connect(ccp, {
        wallet,
        identity: "admin",
        discovery: { enabled: true, asLocalhost: true } 
});
```

However, when I run the following source code, I get the error "Error: No wallet supplied from which to retrieve identity label"

```:
const wallet = await buildWallet(Wallets, walletPath);
await enrollAdmin(this.caClient, wallet, mspOrg1);

const x509 = await wallet.get("admin")
const copy_wallet = await Wallets.newInMemoryWallet();
await copy_wallet.put("admin", x509)

const gateway = new Gateway();
await gateway.connect(ccp, {
        copy_wallet,
        identity: "admin",
        discovery: { enabled: true, asLocalhost: true } 
});
```

When I check the contents, "wallet" and "copy_wallet" look like the same. What's wrong?

**Comment by lindluni** [comment]
Hello @is0356xi,

Discussions are a way to collaborate on the fabric codebase, questions about implementation and bugs should be directed to one of two places:

[RocketChat](https://chat.hyperledger.org)
[Mailing List](fabric@lists.hyperledger.org)

Thank you!
